### ISSUE TITLE
`FileIterator` returns broken URLs when the working directory is `/`

### ISSUE THREAD

**Issue by kkebo** [swiftlang/swift-format#862]
## Description

When you use the `-r` option and `workingDirectory.path` is `"/"`, `String(path.dropFirst(workingDirectory.path.count + 1))` will remove the first character of the file path, so `FileIterator` returns broken URLs and swift-format can't read any file.

https://github.com/swiftlang/swift-format/blob/3ca1a182f2990fb34403fc521b6bd3b839555527/Sources/SwiftFormat/Utilities/FileIterator.swift#L149

## How to reproduce the issue

For instance, on wasm32-unknown-wasi, `URL(fileURLWithPath: ".")` will be `/`, so this is the problem.

And also, the issue can be reproducible on normal platforms like this:

```console
$ pwd
/
$ ls -l | grep *.swift
.rw-r--r--. root root  12 B  Thu Oct 24 02:39:39 2024 test.swift                                                                                                     
$ swift format lint -rs .
<unknown>: error: Unable to open est.swift: file is not readable or does not exist
```

My environment:

- Linux aarch64
- Swift 6.0 development snapshot toolchain (6.0-snapshot-2024-10-08)
  ```
  Swift version 6.0.3-dev (LLVM 81a2745a5c40b5f, Swift 8f95e04bb5d9d9a)
  Target: aarch64-unknown-linux-gnu
  ```
- swift-format bundled with the Swift toolchain

**Comment by ahoppen** [comment]
Synced to Apple’s issue tracker as rdar://138492445

**Comment by kkebo** [comment]
Resolved by #865
